### PR TITLE
Only show private data when in direct message with chatbot

### PR DESCRIPTION
**Pull request by markediez**

### CODE REVIEW

File: dss-chatbot.rb [hunk 1]
```diff
@@ -71,9 +71,11 @@
     self_id = client.self["id"]
     next if data["user"] == self_id
 
+    # True if the channel is one of the channels directly messaging chatbot
+    is_dm = client.ims[data["channel"]] == nil ? false : true
```
Comment (cthielen) [inline dss-chatbot.rb]: Isn't it simpler to say: "is_dm = client.ims[data["channel"]] != nil"?

File: chat_bot_command.rb [hunk 2]
```diff
@@ -34,7 +35,7 @@ def ChatBotCommand.dispatch(message, channel, user)
 
         # If regex matches, run the command
         if regex_match
-          allow_private = is_allowed_private(command_class_reference::TITLE, user.name)
+          allow_private = is_dm ? is_allowed_private(command_class_reference::TITLE, user.name) : false
```
Comment (cthielen) [inline chat_bot_command.rb]: I think it's more readable to say:

allow_private = is_dm && is_allowed_private(command_class_reference::TITLE, user.name)

File: commands/reload.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+module ChatBotCommand
+  class Reload
+    TITLE = "Reload"        # Title of the command
+    REGEX = /^!reload/      # REGEX the command needs to look for
+    COMMAND = "!reload"     # Command to use
+    DESCRIPTION = "Reloads chatbot config files"       # Description of the command
```
Comment (cthielen) [inline commands/reload.rb]: Should be ChatBot, not chatbot.

File: commands/reload.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    # Returns a string indicating if chatbot reloaded its $SETTINGS properly or not
+    # @param message - message posted
+    # @param channel - the channel where the message was posted
+    # @param can_view_private - flag if extra data can be outputted
```
Comment (cthielen) [inline commands/reload.rb]: @param names do not match what's on line 12 e.g. can_view_private versus private_allowed

File: commands/reload.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    def run(message, channel, private_allowed)
+      # Reload sensitive settings from config/*
+      settings_file = $cwd + '/config/settings.yml'
```
Comment (cthielen) [inline commands/reload.rb]: Isn't this already defined in the main file?

File: commands/reload.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      if File.file?(settings_file)
+        $SETTINGS = YAML.load_file(settings_file)
```
Comment (cthielen) [inline commands/reload.rb]: Isn't the loading of the settings file and the checks for needed config options already in the main file? Don't Repeat Yourself

File: commands/reload.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+        if $SETTINGS["GLOBAL"] == nil
```
Comment (cthielen) [inline commands/reload.rb]: We need a way for the !reload command to only work when invoked by certain users. Can you add it to settings as an array list, e.g. RELOAD_USERS: ["cthielen"] ?

File: commands/reload.rb (hunk 6 above)
Comment (markediez) [inline commands/reload.rb]: What if someone does !reload but the they accidentally moved / deleted settings.yml?

Comment (cthielen) [inline commands/reload.rb]: I agree. By Don't Repeat Yourself I mean you should be able to re-use lines 33-48 from the main file verbatim. Make it a callable routine, for instance. Don't copy lines of code.

File: commands/reload.rb (hunk 7 above)
Comment (markediez) [inline commands/reload.rb]: We could use the allow_private architecture. The only downside is that you can only !reload if you're direct messaging chatbot

**Pull request merged**